python/ray/tests/test_state_api.py

```diff
@@ -3598,7 +3599,61 @@ def f(signal):
     all_job_info = client.get_all_job_info()
     assert len(all_job_info) == 1
     assert job_id in all_job_info
-    assert client.get_all_job_info()[job_id].is_running_tasks is True
+    assert all_job_info[job_id].is_running_tasks is True
+
+
+def test_hang_driver_has_no_is_running_task(ray_start_cluster):
+    """
+    When there's a call to JobInfoGcsService.GetAllJobInfo, GCS sends RPC to all drivers
+    for "is_running_task". If the driver is hanging, GCS will give up that RPC and unset
+    the field.
+    """
+    cluster = ray_start_cluster
+    cluster.add_node(num_cpus=10)
+    address = cluster.address
+    ray.init(namespace="test_hang_driver_has_no_is_running_task", address=address)
+
+    # To reliably know a job has a long running task, we need to wait a SignalActor
+    # to know the task has started.
+    signal_actor = SignalActor.options(lifetime="detached", name="signal").remote()
+
+    driver_script = f"""
+import ray
+import time
+
+ray.init(namespace="test_hang_driver_has_no_is_running_task", address="{address}")
+# Now, it's long running...
+@ray.remote
+def long_running():
+    signal_actor = ray.get_actor("signal")
+    ray.get(signal_actor.send.remote())
+    print("begin infinite sleeping...")
+    while True:
+        time.sleep(10000)
+
```

Comment:
There is a simpler way to do this. Check out `RAY_testing_asio_delay_us`.

Reply:
hmm, we want it to "keep running and stuck". If we set RAY_testing_asio_delay_us to inf it won't be able to invoke `signal_actor.send.remote()` in the first place. What about we keep it this way?